### DISCUSSION TITLE
Issues with GitHub REST API for Secret Scanning Alerts

### DISCUSSION THREAD

**Discussion by nmarkana-crest** [community/community#127188] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

REST APIs:
1. [List Secret Scanning Alerts for an Enterprise](https://docs.github.com/en/rest/secret-scanning/secret-scanning?apiVersion=2022-11-28#list-secret-scanning-alerts-for-an-enterprise)
2. [List Secret Scanning Alerts for an Organization
](https://docs.github.com/en/rest/secret-scanning/secret-scanning?apiVersion=2022-11-28#list-secret-scanning-alerts-for-an-organization)

Bugs/Issues:

1. We have a use case to collect secret scanning alerts at particular intervals without duplication. We are interested in receiving the secret scanning alerts in ascending order of the updated field. To avoid data duplication, we are keeping track of the “after” or “page” parameter to fetch alerts. However, we have noticed that whenever we update any alert collected in previous page, in the subsequent pages, the updated alert is received but the some other alerts are lost. This results in data loss for already present alerts, which ideally should not occur. In the REST API for Code Scanning alerts ([link](https://docs.github.com/en/enterprise-cloud@latest/rest/code-scanning/code-scanning?apiVersion=2022-11-28#list-code-scanning-alerts-for-an-organization)), the API itself handles fetching all alerts, including the updated ones, in the upcoming pages.

**Reproduction Steps:**

Let us assume we have 5 alerts (1,2,3,4,5 - all in ascending order of update date) in our GitHub enterprise 

Keeping the per_page to “2” and “sort” ascending on updated date

Expected outcome when no event is updated in between :
1st page - 1,2
2nd page - 3,4
3 page - 5

Expected outcome when any event is updated in between :
1st page - 1,2
Updating the alert 2 from GitHub
2nd page - 3,4
3 page - 5, 2

Actual outcome when any event is updated in between :
1st page - 1,2
Updating the alert 2 from GitHub
2nd page - 4,5
3 page - 2 

In actual outcome, we are not receiving the 3rd event and then that event is lost.

You can perform the same reproduction steps with the code scanning alerts API ([link](https://docs.github.com/en/enterprise-cloud@latest/rest/code-scanning/code-scanning?apiVersion=2022-11-28#list-code-scanning-alerts-for-an-organization)) and verify that we didn’t experience any loss of alerts while updating the previously fetched alerts.

2. In the REST API documentation ([link](https://docs.github.com/en/rest/secret-scanning/secret-scanning?apiVersion=2022-11-28#list-secret-scanning-alerts-for-an-organization)), it is mentioned that there is support for “before” and “after” parameters. However, upon checking the actual response header, there are no URLs with “after” or “before” parameters. Only URLs with the “page” parameters are present.

<img width="1365" alt="Screenshot 2024-06-04 at 2 31 33 PM" src="https://github.com/community/community/assets/163839750/8c01e388-f77c-405f-b58e-37d404ba6179">

Could someone please prioritise looking into this issue as we are currently blocked on it?

**Comment by nmarkana-crest** [comment]
Do we have any updates on this?

**Reply by HyHamza** [reply]
Yes

**Reply by nmarkana-crest** [reply]
@HyByte , if possible can you please share those updates/progress here?

**Reply by HyHamza** [reply]
Nope. sorry
